pynumaflow/batchmapper/async_server.py

```diff
@@ -0,0 +1,213 @@
+
+        # Start the async server
+        await start_async_server(
+            server_new,
+            self.sock_path,
+            self.max_threads,
+            self._server_options,
+            self.server_info_file,
+        )
+
+
+class BatchMapServer(BatchMapAsyncServerBase):
+    """
+    This creates a Server instance to handler an input stream of data and
+    allow data to be streamed back out. The input is not bounded at the SDK level
+    and handler function must be cognizant to handle to handle each message and to decide
+    access patterns.
+
+    Example invocation:
+        from pynumaflow.batchmapper import (
+            Message,
+            Messages,
+            Datum,
+            BatchMapper,
+            BatchMapServer,
+            BatchResponses,
+        )
+
+        class MapperStreamer(BatchMapper):
+            async def handler(self, datum: AsyncIterable[Datum]) -> AsyncIterable[BatchResponses]:
+                async for msg in datum:
+                    msgs = Messages(
+                        Message(value=msv.value, keys=[], tags=[])
+                    )
+                    response = BatchResponses(msg.id, msgs)
+                    yield response
+
+        if __name__ == "__main__":
+            handler = MapperStreamer()
+            grpc_server = BatchMapServer(handler)
+            grpc_server.start()
+    """
+
+    def __init__(
+        self,
+        mapper_instance: MapBatchAsyncCallable,
+        sock_path=MAP_BATCH_SOCK_PATH,
+        max_message_size=MAX_MESSAGE_SIZE,
+        max_threads=MAX_THREADS,
+        server_info_file=MAP_BATCH_SERVER_INFO_FILE_PATH,
+    ):
+        servicer = BatchMapServicer(mapper_instance)
+        super().__init__(
+            servicer,
+            sock_path=sock_path,
+            max_message_size=max_message_size,
+            max_threads=max_threads,
+            server_info_file=server_info_file,
+        )
+
+
+class BatchMapUnaryServer(BatchMapAsyncServerBase):
+    """
+    This creates a server instance to handle a stream of input data and provide it to callers
+    with identical interface to async_mapper's unary interface and functions as a drop-in
+    replacement. Using this may provide a performance boost by utilizing the a GRPC stream
+    rather than true GRPC unary communication to the numa client.
+    """
+
+    def __init__(
+        self,
+        mapper_instance: MapBatchAsyncUnaryCallable,
+        sock_path=MAP_BATCH_SOCK_PATH,
+        max_message_size=MAX_MESSAGE_SIZE,
+        max_threads=MAX_THREADS,
+        server_info_file=MAP_BATCH_SERVER_INFO_FILE_PATH,
+    ):
+        servicer = BatchMapUnaryServicer(mapper_instance)
+        super().__init__(
+            servicer,
+            sock_path=sock_path,
+            max_message_size=max_message_size,
+            max_threads=max_threads,
+            server_info_file=server_info_file,
+        )
+
+
+class BatchMapGroupingServer(BatchMapAsyncServerBase):
```

Comment:
For this functionality, we will keep this controlled through Numa container with the number of messages that are sent to the UDF instead of the SDK buffering them at its end. 

While this does not exist today, We plan to add such a capability in numaflow soon. Once, that's added it should be out of the box configuration for the user where there would be able to specify such a property (of time / num of messages) and we would send the required number itself from numa.

So let's remove this for now